scripts/abi_check.py

```diff
@@ -0,0 +1,240 @@
+                    os.remove(output_path)
+            elif abi_compliance_process.returncode == 1:
+                compliance_return_code = 1
+                self.should_keep_report_dir = True
+                compatibility_report += (
+                    "Compatibility issues found for {}, "
+                    "for details see {}\n".format(mbed_module, output_path)
+                )
+            else:
+                raise Exception(
+                    "abi-compliance-checker failed with a return code of {},"
+                    " aborting".format(abi_compliance_process.returncode)
+                )
+            os.remove(self.old_dumps[mbed_module])
+            os.remove(self.new_dumps[mbed_module])
+        if not self.should_keep_report_dir and not self.keep_all_reports:
+            os.rmdir(self.report_dir)
+        self.log.info(compatibility_report)
+        return compliance_return_code
+
+    def check_for_abi_changes(self):
+        self.check_repo_path()
+        self.check_abi_tools_are_installed()
+        self.old_dumps = self.get_abi_dump_for_ref(self.old_rev)
+        self.new_dumps = self.get_abi_dump_for_ref(self.new_rev)
+        return self.get_abi_compatibility_report()
+
+
+def run_main():
+    try:
+        parser = argparse.ArgumentParser(
+            description=(
+                "This script is a small wrapper around the "
+                "abi-compliance-checker and abi-dumper tools, applying them "
+                "to compare the ABI and API of the library files from two "
+                "different Git revisions within an Mbed TLS repository."
+                " The results of the comparison are formatted as HTML and"
+                " stored at a configurable location. Returns 0 on success, "
+                "1 on ABI/API non-compliance, and 2 if there is an error "
+                "while running the script. # Note: must be run from "
+                "Mbed TLS root."
+            )
+        )
+        parser.add_argument(
+            "-r", "--report_dir", type=str, default="reports",
+            help="directory where reports are stored, default is reports",
+        )
+        parser.add_argument(
+            "-k", "--keep_all_reports", action="store_true",
+            help="keep all reports, even if there are no compatibility issues",
+        )
+        parser.add_argument(
+            "-o", "--old_rev", type=str, help="revision for old version",
+            required=True
+        )
+        parser.add_argument(
+            "-n", "--new_rev", type=str, help="revision for new version",
+            required=True
+        )
+        abi_args = parser.parse_args()
+        abi_check = AbiChecker(
+            abi_args.report_dir, abi_args.old_rev,
+            abi_args.new_rev, abi_args.keep_all_reports
+        )
+        return_code = abi_check.check_for_abi_changes()
+        sys.exit(return_code)
+    except Exception as error:
+        traceback.print_exc(error)
```

Comment:
```traceback.print_exc``` does not take exception as argument. It's first argument is a default argument ```limit```. See https://docs.python.org/2/library/traceback.html#traceback.print_exc

This will raise it's own exception masking the real one.

Reply:
Fixed.